portainer/app/executor.py

```diff
@@ -227,7 +227,7 @@ def _build_image(self, driver, taskInfo, buildTask):
                             str("%s: %s" % (image_name, message))
                         )
             else:
-                sandbox_dir = os.environ["MESOS_DIRECTORY"]
+                sandbox_dir = os.environ.get("MESOS_SANDBOX", os.environ["MESOS_DIRECTORY"])
```

Comment:
I opened [MESOS-2612](https://issues.apache.org/jira/browse/MESOS-2612) a little while ago and will submit a patch for it in the next day or so, I wouldn't suggest changing this code but would be nice to keep things consistent on the mesos side.